src/test/java/org/opensearch/neuralsearch/query/HybridQueryIT.java

```diff
@@ -692,6 +708,192 @@ public void testWrappedQueryWithFilter_whenIndexAliasHasFilters_thenSuccess() {
         }
     }
 
+    @SneakyThrows
+    public void testInnerHits_whenNestedFieldsInDocuments_thenSuccess() {
+        try {
+            initializeIndexIfNotExist(TEST_MULTI_DOC_INDEX_WITH_NESTED_TYPE_NAME_ONE_SHARD);
+            createSearchPipelineWithResultsPostProcessor(SEARCH_PIPELINE);
+
+            NestedQueryBuilder nestedQueryBuilder = new NestedQueryBuilder(
+                TEST_NESTED_TYPE_FIELD_NAME_1,
+                QueryBuilders.boolQuery()
+                    .should(QueryBuilders.termQuery(TEST_NESTED_TYPE_FIELD_NAME_1 + "." + NESTED_FIELD_1, NESTED_FIELD_1_VALUE)),
+                ScoreMode.Total
+            ).innerHit(
+                new InnerHitBuilder().setStoredFieldNames(Collections.singletonList("_none_"))
+                    .setFetchSourceContext(new FetchSourceContext(false))
+            );
+
+            TermQueryBuilder termQueryBuilder = QueryBuilders.termQuery(TEST_TEXT_FIELD_NAME_1, TEST_QUERY_TEXT3);
+            HybridQueryBuilder hybridQueryBuilderOnlyTerm = new HybridQueryBuilder();
+            hybridQueryBuilderOnlyTerm.add(nestedQueryBuilder);
+            hybridQueryBuilderOnlyTerm.add(termQueryBuilder);
+
+            Map<String, Object> searchResponseAsMap = search(
+                TEST_MULTI_DOC_INDEX_WITH_NESTED_TYPE_NAME_ONE_SHARD,
+                hybridQueryBuilderOnlyTerm,
+                null,
+                10,
+                Map.of("search_pipeline", SEARCH_PIPELINE)
+            );
+
+            assertEquals(2, getHitCount(searchResponseAsMap));
+            assertTrue(getMaxScore(searchResponseAsMap).isPresent());
+            assertEquals(0.5f, getMaxScore(searchResponseAsMap).get(), DELTA_FOR_SCORE_ASSERTION);
+
+            Map<String, Object> total = getTotalHits(searchResponseAsMap);
+            assertNotNull(total.get("value"));
+            assertEquals(2, total.get("value"));
+            assertNotNull(total.get("relation"));
+            assertEquals(RELATION_EQUAL_TO, total.get("relation"));
+
+            // check for inner hits, first such hit is empty second has one hit
+            List<Map<String, Object>> hits = getNestedHits(searchResponseAsMap);
+            assertNotNull(hits);
+            assertEquals(2, hits.size());
+
+            Map<String, Object> hitOne = hits.get(0);
```

Comment:
let me think, maybe I can refactor few smaller methods. But generally it's neeeded because we're checking the internal structure of the "hit" object, and code cannot be reused between those two sections as one it asserting hit with result, and other is asserting empty hit

Reply:
I've pushed a commit with refactoring of that code, able to extract one private method that can be re-used in assertion